File: swift/functions/string_manipulation.swift
#!/usr/bin/env swift

let text = "  Hello, World!  "

print("Original: '\(text)'")
print("Upper: '\(text.uppercased())'")
print("Lower: '\(text.lowercased())'")
print("Trim: '\(text.trimmingCharacters(in: .whitespaces))'")
print("Replace: '\(text.replacingOccurrences(of: "World", with: "Swift"))'")
print("Split: \(text.trimmingCharacters(in: .whitespaces).split(separator: ","))")
print("Contains: \(text.contains("World"))")
print("HasPrefix: \(text.trimmingCharacters(in: .whitespaces).hasPrefix("Hello"))")
print("HasSuffix: \(text.trimmingCharacters(in: .whitespaces).hasSuffix("!"))")
print("Length: \(text.count)")
print("Substring: '\(text.trimmingCharacters(in: .whitespaces).prefix(5))'")

File: go/functions/string_manipulation.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	text := "  Hello, World!  "

	fmt.Printf("Original: '%s'\n", text)
	fmt.Printf("Upper: '%s'\n", strings.ToUpper(text))
	fmt.Printf("Lower: '%s'\n", strings.ToLower(text))
	fmt.Printf("Trim: '%s'\n", strings.TrimSpace(text))
	fmt.Printf("Replace: '%s'\n", strings.Replace(text, "World", "Go", -1))
	fmt.Printf("Split: %v\n", strings.Split(strings.TrimSpace(text), ", "))
	fmt.Printf("Contains: %v\n", strings.Contains(text, "World"))
	fmt.Printf("HasPrefix: %v\n", strings.HasPrefix(strings.TrimSpace(text), "Hello"))
	fmt.Printf("HasSuffix: %v\n", strings.HasSuffix(strings.TrimSpace(text), "!"))
	fmt.Printf("Repeat: '%s'\n", strings.Repeat("-", 20))
}
